### DISCUSSION TITLE
Recommended way to specify release for `sentry_dart_plugin` when uploading debug symbols?

### DISCUSSION THREAD

**Discussion by sumanthratna** [getsentry/sentry-dart#2234] [Q&A]
I'm building my Flutter app for iOS using `flutter build ipa --release --dart-define "TAIKI_API_ENDPOINT=..." --dart-define "SENTRY_DSN=..." --dart-define "SENTRY_RELEASE=..." --dart-define "SENTRY_ENVIRONMENT=release" --obfuscate --split-debug-info=./debug-info`. Then, I upload the debug symbols by running `dart run sentry_dart_plugin`. 

`sentry_dart_plugin` computes the release to be `app@1.0.0+1`. Am I supposed to hard-code the `SENTRY_RELEASE=` when running `flutter build` before `sentry_dart_plugin`? Is there a convenient way to run both of these steps such that I don't need to manually specify the release when building, while still having the debug symbols and built app point to the same Sentry release?

**Comment by buenaflor** [comment]
Hey, the plugin in this case doesn't know anything about the release you set during the build process so by default it uses the version specificed in `puspec.yaml` I assume.

What you can also do is update the puspec version, the sentry flutter sdk reads it automatically and sets it as release.

So both plugin and flutter have access to pubspec.yaml

**Comment by dogs342** [comment]
You can automate the release version for both the app build and the debug symbol upload by dynamically setting the `SENTRY_RELEASE` during the build process. You can extract the version from `pubspec.yaml` or use a custom script to pass it to both `flutter build ipa` and `sentry_dart_plugin`. This ensures that both the app and debug symbols are linked to the same Sentry release without hardcoding it manually.

Also, consider using the `[latestdogs](https://doglatest.com/)` flag to streamline integration if it's supported by the plugin.
